File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string filtered = "";
        for(char x:s){
            if(isalnum(x))
                filtered+=tolower(x);
        }
        int n = filtered.size();
            for(int i=0;i<n/2;i++){
                if(filtered[i]!=filtered[n-1-i]) return false;
            }
            return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        for(char c : s.toCharArray()){
            if(Character.isLetter(c) || Character.isDigit(c)){
                sb.append(Character.toLowerCase(c));
            }
        }
        String str = sb.toString();
        int left = 0 , right = str.length() - 1;
        while(left <= right){
            if(str.charAt(left) != str.charAt(right)) return false;
            else {
                left++ ;
                right--;
            }
        }
        return true;
    }
}